Add educational code pointers to top level README (#457)
pytorch/torchtitan: README.md

[![Welcome to torchtitan!](assets/images/titan_play_video.png)](https://youtu.be/ee5DOEqD35I?si=_B94PbVv0V5ZnNKE "Welcome to torchtitan!")

> ### Dive Into the code

Reviewer: ```suggestion
### Dive into the code
```

You may want to see how the model is defined or how parallelism techniques are applied. For a guided tour, see these files first:
* [train.py](https://github.com/pytorch/torchtitan/blob/main/train.py) - the main training loop and high-level setup code

> * [torchtitan/parallelisms/parallelize_llama.py](https://github.com/pytorch/torchtitan/blob/main/torchtitan/parallelisms/parallelize_llama.py) - helpers for applying TP/DP/PP parallelisms to the model

Reviewer: nit: 
```suggestion
* [torchtitan/parallelisms/parallelize_llama.py](https://github.com/pytorch/torchtitan/blob/main/torchtitan/parallelisms/parallelize_llama.py) - helpers for applying Data / Tensor / Pipeline Parallelisms to the model
```

Reviewer 2: https://github.com/pytorch/torchtitan/blob/07ab2f9a7298c93f3dd67b728c0c25c6c92292e2/torchtitan/parallelisms/parallelize_llama.py#L7-L8

The file also includes non-parallelism related techniques like AC and `torch.compile`. Are these worth mentioning in the readme?

* [torchtitan/checkpoint.py](https://github.com/pytorch/torchtitan/blob/main/torchtitan/checkpoint.py) - utils for saving/loading distributed checkpoints

> * [torchtitan/models/llama/model.py](https://github.com/pytorch/torchtitan/blob/main/torchtitan/models/llama/model.py) - the LLaMa model definition (shared for llama2/llama3 variants)

Reviewer: ```suggestion
* [torchtitan/models/llama/model.py](https://github.com/pytorch/torchtitan/blob/main/torchtitan/models/llama/model.py) - the Llama model definition (shared for Llama2 and Llama3 variants)
```
